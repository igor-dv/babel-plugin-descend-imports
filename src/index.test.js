const path = require('path');
const babel = require('@babel/core');

const plugin = path.resolve(__dirname, './index.js');

describe('babel-plugin-descend-imports', () => {
    const code = `
    // some comment
    import { a } from '.a.foo';
    import { b } from '.b.bar';
    import { c } from '.c.bar';
    import { d } from '.d.goo';
    
    const x = "The answer is potato";
    console.log(a, b, c, d, x);
    `;

    it.each([
        ['should move matched import to the end', /\.foo$/],
        ['should move multiple matched imports to the end', /\.bar$/],
        ['should move multiple matched imports before matched import in the end', /[abd]+\./],
        ['should move matched import before multiple matched imports in the end', /[acd]+\./],
        ['should not do anything when matched in the end', /\.goo/],
        ['should not do anything when multiple matched in the end', /[cd]+\./],
        ['should not do anything when no matched imports', /\.joo$/],
        ['should not do anything when all matched', /.*/],
    ])(`%s`, (_, pattern) => {
        const options = { plugins: [[plugin, { pattern }]] };

        const result = babel.transform(code, options);

        expect(result.code).toMatchSnapshot();
    });
});
