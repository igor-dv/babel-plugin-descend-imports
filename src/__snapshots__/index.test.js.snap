// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`babel-plugin-descend-imports should move matched import before multiple matched imports in the end 1`] = `
"// some comment
import { b } from '.b.bar';
import { a } from '.a.foo';
import { c } from '.c.bar';
import { d } from '.d.goo';
const x = \\"The answer is potato\\";
console.log(a, b, c, d, x);"
`;

exports[`babel-plugin-descend-imports should move matched import to the end 1`] = `
"// some comment
import { b } from '.b.bar';
import { c } from '.c.bar';
import { d } from '.d.goo';
import { a } from '.a.foo';
const x = \\"The answer is potato\\";
console.log(a, b, c, d, x);"
`;

exports[`babel-plugin-descend-imports should move multiple matched imports before matched import in the end 1`] = `
"// some comment
import { c } from '.c.bar';
import { a } from '.a.foo';
import { b } from '.b.bar';
import { d } from '.d.goo';
const x = \\"The answer is potato\\";
console.log(a, b, c, d, x);"
`;

exports[`babel-plugin-descend-imports should move multiple matched imports to the end 1`] = `
"// some comment
import { a } from '.a.foo';
import { d } from '.d.goo';
import { b } from '.b.bar';
import { c } from '.c.bar';
const x = \\"The answer is potato\\";
console.log(a, b, c, d, x);"
`;

exports[`babel-plugin-descend-imports should not do anything when all matched 1`] = `
"// some comment
import { a } from '.a.foo';
import { b } from '.b.bar';
import { c } from '.c.bar';
import { d } from '.d.goo';
const x = \\"The answer is potato\\";
console.log(a, b, c, d, x);"
`;

exports[`babel-plugin-descend-imports should not do anything when matched in the end 1`] = `
"// some comment
import { a } from '.a.foo';
import { b } from '.b.bar';
import { c } from '.c.bar';
import { d } from '.d.goo';
const x = \\"The answer is potato\\";
console.log(a, b, c, d, x);"
`;

exports[`babel-plugin-descend-imports should not do anything when multiple matched in the end 1`] = `
"// some comment
import { a } from '.a.foo';
import { b } from '.b.bar';
import { c } from '.c.bar';
import { d } from '.d.goo';
const x = \\"The answer is potato\\";
console.log(a, b, c, d, x);"
`;

exports[`babel-plugin-descend-imports should not do anything when no matched imports 1`] = `
"// some comment
import { a } from '.a.foo';
import { b } from '.b.bar';
import { c } from '.c.bar';
import { d } from '.d.goo';
const x = \\"The answer is potato\\";
console.log(a, b, c, d, x);"
`;
